ci/scripts/util_enable_core_dumps.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# NOTE: this script is not marked executable as it should be source'd
+# for `ulimit` to take effect.
+
+set -ex
+
+platform=$(uname)
+
+if [ "${platform}" = "Linux" ]; then
+  # We need to override `core_pattern` because
+  # 1. the original setting may reference apport, which is not available under
+  #    most Docker containers;
+  # 2. we want to write the core file in a well-known directory.
+  sudo sysctl -w kernel.core_pattern="/tmp/core.%e.%p"
```

Comment:
Can we keep `core.%e.%p`?

`cpp/build-support/run-test.sh` assumes it:
https://github.com/apache/arrow/blob/c455d6b8c4ae2cb22baceb4c27e1325b973d39e1/cpp/build-support/run-test.sh#L119-L129

Reply:
Well, does this function work? Is it used anywhere?

Reply:
Recently, this doesn't work... If this works, we can see baktrace from CI logs.

In general, our tests are run via this script:
https://github.com/apache/arrow/blob/b2e0668a7230131a35e228958aa895442f45152a/cpp/cmake_modules/BuildUtils.cmake#L771-L776

Reply:
Can I just update `cpp/build-support/run-test.sh` to look into `/tmp` instead of the current directory?

Reply:
Yes.

If we use this approach, could you also update the comment in `docker-compose.yml`?
https://github.com/apache/arrow/blob/989508418b59e19037cbc58b46fb72ec2adf6234/docker-compose.yml#L29-L54